script.js

```diff
@@ -24,7 +24,27 @@ function makePageForEpisodes(episodeList) {
 
     div.append(h3, img, p);
     rootElem.appendChild(div);
+
+    const label = document.querySelector("label");
+    label.textContent = `Displaying ${getAllEpisodes().length}/${getAllEpisodes().length} episodes`;
   })
 }
 
 window.onload = setup;
+
+function searchEpisode (value) {
+  const rootElem = document.getElementById("root");
+  rootElem.replaceChildren();
+
+  const label = document.querySelector("label");
+
+  const filteredEpisodes = getAllEpisodes().filter(episode =>
+    episode.name.toLowerCase().includes(value.toLowerCase()) || episode.summary.toLowerCase().includes(value.toLowerCase())
+  );
+
+  return makePageForEpisodes(filteredEpisodes),
+  label.textContent = `Displaying ${filteredEpisodes.length}/${getAllEpisodes().length} episodes`;
+}
+
+const input = document.querySelector("#input");
+input.addEventListener("input", () => searchEpisode(input.value));
```

Comment:
usually we use one of these event names when reacting on user input: 
- https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent#events 
- https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events
if you take a very close look at "input" event you might notice some strange behaviour while editing text. 
For example, "input" event works for any key pressed. but what if you want to change it's behaviour and only search when "Enter" key pressed by the user?